recipes/libcugraph/build.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+# Derived from cugraph build script, as seen here:
+# https://github.com/rapidsai/cugraph/blob/db20b485cfc5399214afcff604b38493f38e83bf/build.sh#L137
+
+# NOTE: it is assumed the RMM header-only sources have been downloaded from the
+# multi-source "source:" section in the meta.yaml file.
+# cmake must be able to find the RMM headers using find_path(). The RMM_ROOT env
+# var is set so RMM_ROOT/include results in a valid dir for cmake to search.
+
+export CUGRAPH_SRC_DIR="${SRC_DIR}/cugraph"
+export RMM_ROOT="${SRC_DIR}/rmm"
+export LIBCUGRAPH_BUILD_DIR=${LIBCUGRAPH_BUILD_DIR:=${CUGRAPH_SRC_DIR}/cpp/build}
+export GPU_ARCH=ALL
+export INSTALL_PREFIX=${PREFIX:=${CONDA_PREFIX}}
+export BUILD_DISABLE_DEPRECATION_WARNING=ON
+export BUILD_TYPE=Release
+export BUILD_CPP_TESTS=OFF
+export BUILD_CPP_MG_TESTS=OFF
+export BUILD_STATIC_FAISS=OFF
+export PARALLEL_LEVEL=${CPU_COUNT}
+export INSTALL_TARGET=install
+export VERBOSE_FLAG=""
+
+mkdir -p ${LIBCUGRAPH_BUILD_DIR}
+cd ${LIBCUGRAPH_BUILD_DIR}
+cmake -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX} \
+      ${GPU_ARCH} \
+      -DDISABLE_DEPRECATION_WARNING=${BUILD_DISABLE_DEPRECATION_WARNING} \
```

Comment:
Maybe this would help?

```suggestion
      -DDISABLE_DEPRECATION_WARNING=${BUILD_DISABLE_DEPRECATION_WARNING} \
      -DTHREADS_PREFER_PTHREAD_FLAG=TRUE \
```

https://cmake.org/cmake/help/latest/module/FindThreads.html

Reply:
The output in the log you linked above was the last entry in the CMakeError.log, which was populated as cmake was attempting to find libpthreads. It later correctly found libpthread on it's own:
![image](https://user-images.githubusercontent.com/3039903/114953636-64a9f700-9e1e-11eb-959c-81c64f6fe4cd.png)
I think dumping the CMakeError.log was confusing here since it made it look like the root cause was related to the (expected) failed libpthreads check.  I've removed the line that dumps these logs, but maybe I can add formatting to make it more obvious that the last output of cmake to start with will be above the log dumps.